.github/workflows/ci-build.yaml

```diff
@@ -129,72 +132,75 @@ jobs:
-      - uses: actions/download-artifact@v3
-        name: Download argoexec image
+      - name: Download argoexec image
+        uses: actions/download-artifact@v3
         with:
           name: argoexec
           path: /tmp
-      - run: docker load < /tmp/argoexec_image.tar
-        name: Load argoexec image
+      - name: Load argoexec image
+        run: docker load < /tmp/argoexec_image.tar
       - name: Set-up /etc/hosts
         run: |
           echo '127.0.0.1 dex'      | sudo tee -a /etc/hosts
           echo '127.0.0.1 minio'    | sudo tee -a /etc/hosts
           echo '127.0.0.1 postgres' | sudo tee -a /etc/hosts
           echo '127.0.0.1 mysql'    | sudo tee -a /etc/hosts
           echo '127.0.0.1 azurite'  | sudo tee -a /etc/hosts
-      - run: make install PROFILE=${{matrix.profile}} STATIC_FILES=false
-        name: Install manifests
-      - run: make controller kit STATIC_FILES=false
-        name: Build controller
-      - run: make cli STATIC_FILES=false
+      - name: Install manifests
+        run: make install PROFILE=${{matrix.profile}} STATIC_FILES=false
+      - name: Build controller
+        run: make controller kit STATIC_FILES=false
+      - name: Build CLI
+        run: make cli STATIC_FILES=false
         if: ${{matrix.test == 'test-api' || matrix.test == 'test-cli' || matrix.test == 'test-java-sdk' || matrix.test == 'test-python-sdk'}}
-        name: Build CLI
-      - run: make start PROFILE=${{matrix.profile}} AUTH_MODE=client STATIC_FILES=false LOG_LEVEL=info API=${{matrix.test == 'test-api' || matrix.test == 'test-cli' || matrix.test == 'test-java-sdk' || matrix.test == 'test-python-sdk'}} UI=false > /tmp/argo.log 2>&1 &
-        name: Start controller/API
-      - run: make wait API=${{matrix.test == 'test-api' || matrix.test == 'test-cli' || matrix.test == 'test-java-sdk' || matrix.test == 'test-python-sdk'}}
+      - name: Start controller/API
+        run: make start PROFILE=${{matrix.profile}} AUTH_MODE=client STATIC_FILES=false LOG_LEVEL=info API=${{matrix.test == 'test-api' || matrix.test == 'test-cli' || matrix.test == 'test-java-sdk' || matrix.test == 'test-python-sdk'}} UI=false > /tmp/argo.log 2>&1 &
+      - name: Wait for controller to be up
+        run: make wait API=${{matrix.test == 'test-api' || matrix.test == 'test-cli' || matrix.test == 'test-java-sdk' || matrix.test == 'test-python-sdk'}}
         timeout-minutes: 5
-        name: Wait for controller to be up
       - name: Run tests ${{matrix.test}}
         run: make ${{matrix.test}} E2E_SUITE_TIMEOUT=20m STATIC_FILES=false
-      - if: ${{ failure() }}
-        name: MinIO/MySQL deployment
+
+      # failure debugging below
+      - name: Failure debug - get MinIO/MySQL deployment
+        if: ${{ failure() }}
         run: |
           set -eux
           kubectl get deploy
           kubectl describe deploy
-      - if: ${{ failure() }}
-        name: MinIO/MySQL pods
+      - name: Failure debug - get MinIO/MySQL pods
```

Comment:
This is too implementation specific and may change

Reply:
Not sure what you mean, it already said "MinIO/MySQL pods", which I had no clue why it was doing that before, so I added "Failure debug" to be explicit.
I also didn't know what "MinIO/MySQL pods" meant. Versus, I run an SRE team and if you say "`[kubectl] get` a Pod", everyone knows what means. If you say "MinIO/MySQL pods", I would say "what about them?".
I can change it to `describe` which would be even more explicit.

There's two lines of code here, so yes, I would very much expect that the name would need changing if the two lines change. I don't think that's a bad thing, that means the name is explicit about what the two lines do and there is no guesswork to be had when looking at the CI logs, which normally auto-collapse these under the name.

Reply:
I updated it to `describe`